src/tests/example.spec.js

```diff
@@ -158,4 +158,79 @@ test.describe('Saucedemo app basic tests', () => {
             expect(cartPrice).toBe(selectedProducts[i].price);
         }
     });
+
+    test('check info about several added random products and Total Price on the "Checkout: Overview" page', async (
+        /** @type {{ app: import('../pages/Application').Application }} */{ app },
+    ) => {
+        // get all the elements of the products
+        const products = await app.inventory.product;
+        const productCount = await products.count();
+
+        // generate three unique indexes
+        const randomIndexes = getUniqueIndexes(productCount);
+        // const randomIndexes = new Set();
+        // const uniqueIndexesCount = Math.min(3, productCount);
+        // while (randomIndexes.size < uniqueIndexesCount) {
+        //     const randomIndex = Math.floor(Math.random() * productCount);
+        //     randomIndexes.add(randomIndex);
+        // }
+
+        // store information about selected products
+        const selectedProducts = await app.inventory.addedItemToCartInfoByIndex(randomIndexes);
+
+        // go to the shopping cart
+        await app.inventory.shoppingCartClick();
+
+        // Checkout: Add Information
+        await app.shoppingCart.checkoutButtonClick();
+
+        // filling out the form
+        await app.checkoutStepOne.fillingOutFormWithUserData();
+
+        // check every product on the "Checkout: Overview" page
+        const checkoutItems = await app.checkoutStepTwo.cartItems;
+
+        for (let i = 0; i < selectedProducts.length; i++) {
+            const checkoutItem = checkoutItems.nth(i);
+
+            const checkoutName = await checkoutItem.getByTestId('inventory-item-name').textContent();
+            const checkoutDescription = await checkoutItem.getByTestId('inventory-item-desc').textContent();
+            const checkoutPrice = await checkoutItem.getByTestId('inventory-item-price').textContent();
+
+            expect(checkoutName).toBe(selectedProducts[i].name);
+            expect(checkoutDescription).toBe(selectedProducts[i].description);
+            expect(checkoutPrice).toBe(selectedProducts[i].price);
+        }
+
+        // verify calculated Total Price
+        // get all prices
+        const allPrices = await app.checkoutStepTwo.getAllPrices();
+
+        const numbersAllPrices = allPrices
+            .map((price) => {
+                const priceWithoutDollar = price.slice(1);
+                return Number(priceWithoutDollar);
+            });
+
+        // calculate the expected ItemTotal, TaxAmount, TotalPrice
+        let expectedItemTotal = 0;
+        for (let number of numbersAllPrices) {
+            expectedItemTotal += number;
+        };
```

Comment:
Same here. You can move it to a helper. 
Name it `calculateItemTotal` or something like that. You can pass here your `allPrices` and under the hood it will use your another helper which you create (`parsePrices`) and return calculated amount. 
So you can hide all the logic from lines 209-219 inside one line`const expectedItemTotal = calculateItemTotal(allPrices)`